Add support for public keys when encrypting images

mem opened the pull request:
Extend the existing SIF file encryption so that it's possible to specify
a PEM file with a public key instead of a plain text key. When the
public key is specified, a secret will be automatically generated and it
will be encrypted with the specified public key.

Signed-off-by: Marcelo E. Magallon <marcelo@sylabs.io>

**Description of the Pull Request (PR):**

Write your description of the PR here. Be sure to include as much background,
and details necessary for the reviewers to understand exactly what this is
fixing or enhancing.


**This fixes or addresses the following GitHub issues:**

- Fixes #


**Before submitting a PR, make sure you have done the following:**

- Read the [Guidelines for Contributing](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md), and this PR conforms to the stated requirements.
- Added changes to the [CHANGELOG](https://github.com/sylabs/singularity/blob/master/CHANGELOG.md) if necessary according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added tests to validate this PR and tested this PR locally with a `make testall`
- Based this PR against the appropriate branch according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added myself as a contributor to the [Contributors File](https://github.com/sylabs/singularity/blob/master/CONTRIBUTORS.md)


Attn: @singularity-maintainers

Commit:
Add support for public keys when encrypting images

Extend the existing SIF file encryption so that it's possible to specify
a PEM file with a public key instead of a plain text key. When the
public key is specified, a secret will be automatically generated and it
will be encrypted with the specified public key.

Signed-off-by: Marcelo E. Magallon <marcelo@sylabs.io>

schebro commented on internal/pkg/util/fs/mount/mount_linux.go:
```diff
@@ -306,13 +306,15 @@ func GetSizeLimit(options []string) (uint64, error) {
 	for _, opt := range options {
 		if strings.HasPrefix(opt, "key=") {
-			return strings.TrimPrefix(opt, "key="), nil
+			// TODO(mem): base64 encode the key?
```
You might need to base64 encode here. Otherwise, "," or "." in the key may create problems.

mem replied:
Thanks, fixed!

schebro reviewed (changes requested):
Looks good other than the encoding comment. I can start from this and work on private key decryption.

Commit:
Use []byte instead of string for passing keys around

In Go a `string` represents a UTF8 string. Since we need to support
random sequences of bytes, which might not be legal UTF8 strings, change
the key from string to []byte.

Signed-off-by: Marcelo E. Magallon <marcelo@sylabs.io>

Commit:
Support opening images using a private PEM key

Add support for opening (run, exec, etc) images that contain an
encrypted decryption key. The decryption key is encrypted using a public
key in PEM format and it's decrypted using a private key in PEM format.

For now, assume that the first Generic partition linking to the system
partition contains the encrypted decryption key.

Signed-off-by: Marcelo E. Magallon <marcelo@sylabs.io>
